### PR TITLE
Adding details to Conformance Tests using RFC 2119 standards.

### PR DESCRIPTION
**Pull request by brahmaroutu**
This PR is part of the conformance documentation. This is to provide more formal specification using RFC 2119 keywords to describe the test so that who ever is running conformance tests do not have to go through the code to understand why and what is tested.
The documentation information added here into each of the tests eventually result into a document which is currently checked in at location https://github.com/cncf/k8s-conformance/blob/master/docs/KubeConformance-1.9.md

I would like to have this PR reviewed for v1.10 as I consider it important to strengthen the conformance documents.

### CODE REVIEW

**Comment by brahmaroutu** [comment]
@WilliamDenniss @duglin @bradtopol Please review conformance documentation.

**Comment by brahmaroutu** [comment]
@sttts can you please review.

File: test/e2e/apimachinery/custom_resource_definition.go [hunk 1]
```diff
@@ -34,9 +34,9 @@ var _ = SIGDescribe("CustomResourceDefinition resources", func() {
 
 	Context("Simple CustomResourceDefinition", func() {
 		/*
-			   	   Testname: crd-creation-test
-			   	   Description: Create a random Custom Resource Definition and make sure
-				   the API returns success.
+			Release : v1.9
+			Testname: Customer Resource Definition, create
```
Comment (duglin) [inline test/e2e/apimachinery/custom_resource_definition.go]: s/Customer/Custom/

File: test/e2e/apimachinery/custom_resource_definition.go [hunk 2]
```diff
@@ -34,9 +34,9 @@ var _ = SIGDescribe("CustomResourceDefinition resources", func() {
...
+			Description: Create a API extension client, define a random custom resource definition, create the custom resource. API server MUST be able to create the customer resource.
```
Comment (duglin) [inline test/e2e/apimachinery/custom_resource_definition.go]: s/customer resource/custom resource/

File: test/e2e/apimachinery/garbage_collector.go [hunk 3]
```diff
@@ -515,10 +513,9 @@ var _ = SIGDescribe("Garbage collector", func() {
 	})
 
 	/*
-		    Testname: garbage-collector-delete-deployment-propagation-background
-		    Description: Ensure that if deleteOptions.PropagationPolicy is set to Background,
-			then deleting a Deployment should cause ReplicaSets created
-			by that Deployment to also be deleted.
+		Release : v1.9
+		Testname: Garbage Collector, delete deployment,  propagation policy background
+		Description: Create a deployment with a replicaset. Once replicaset is created , delete the deployment  with deleteOptions.PropagationPolicy set to Background. Deleting the deployment MUST delete the replicaset created by the deployment.
```
Comment (duglin) [inline test/e2e/apimachinery/garbage_collector.go]: What happens to the Pods in this case? Are they orphaned or GC'd too?

File: test/e2e/apimachinery/garbage_collector.go [hunk 4]
```diff
@@ -571,10 +568,9 @@ var _ = SIGDescribe("Garbage collector", func() {
 	})
 
 	/*
-		    Testname: garbage-collector-delete-deployment-propagation-true
-		    Description: Ensure that if deleteOptions.PropagationPolicy is set to Orphan,
-			then deleting a Deployment should cause ReplicaSets created
-			by that Deployment to be orphaned.
+		Release : v1.9
+		Testname: Garbage Collector, delete deployment, propagation policy orphan
+		Description: Create a deployment with a replicaset. Once replicaset is created , delete the deployment  with deleteOptions.PropagationPolicy set to Orphan. Deleting the deployment MUST cause the replicaset created by the deployment to be orphaned.
```
Comment (duglin) [inline test/e2e/apimachinery/garbage_collector.go]: Should state the expect behavior of the Pods too.

File: test/e2e/apimachinery/garbage_collector.go [hunk 5]
```diff
@@ -729,9 +725,9 @@ var _ = SIGDescribe("Garbage collector", func() {
 
 	// TODO: this should be an integration test
 	/*
-		    Testname: garbage-collector-multiple-owners
-		    Description: Ensure that if a Pod has multiple valid owners, it will not be deleted
-			when one of of those owners gets deleted.
+		Release : v1.9
+		Testname: Garbage Collector, multiple owners
+		Description: Create a replication controller RC1, with maximum allocatable Pods between 10 and 100 replicas. Create second replication controller RC2 and set RC2 as owner for half of those replicas. Once RC1 is created and the all Pods are created, delete RC1 with deleteOptions.PropagationPolicy set to Foreground. Half of the Pods that has RC2 as owner SHOULD not be deleted but have a deletion timestamp. Deleting the Replication Controller MUST not delete Pods that are owned by multiple replication controllers.
```
Comment (duglin) [inline test/e2e/apimachinery/garbage_collector.go]: I think the SHOULD here should be a MUST, no?

File: test/e2e/apimachinery/garbage_collector.go [hunk 6]
```diff
@@ -843,9 +839,9 @@ var _ = SIGDescribe("Garbage collector", func() {
 
 	// TODO: should be an integration test
 	/*
-		    Testname: garbage-collector-dependency-cycle
-		    Description: Ensure that a dependency cycle will
-			not block the garbage collector.
+		Release : v1.9
+		Testname: Garbage Collector, dependency cycle
+		Description: Create three pods, patch them with Owner references such that pod1 has pod3, pod2 has pod1 and pod3 has pod2 as owner references respectively. Delete pod1 SHOULD delete all pods. The dependency cycle SHOULD not block the garbage collection.
```
Comment (duglin) [inline test/e2e/apimachinery/garbage_collector.go]: s/SHOULD/MUST/ in both spots, I think

File: test/e2e/apimachinery/garbage_collector.go (hunk 4 above)
Comment (brahmaroutu) [reply test/e2e/apimachinery/garbage_collector.go]: Test is only testing for RS. Pods can be tested too.

File: test/e2e/apimachinery/garbage_collector.go (hunk 6 above)
Comment (brahmaroutu) [reply test/e2e/apimachinery/garbage_collector.go]: @duglin Thanks for the review, I fixed as per your comments.

**Review by brahmaroutu — COMMENTED** [review]
Fixed.

File: test/e2e/apimachinery/garbage_collector.go (hunk 4 above)
Comment (duglin) [reply test/e2e/apimachinery/garbage_collector.go]: My point is that we need to state what the expected behavior for the pods are. In other words make it clear what people should expect to see.

File: test/e2e/apimachinery/garbage_collector.go (hunk 3 above)
Comment (brahmaroutu) [reply test/e2e/apimachinery/garbage_collector.go]: Fixed.

File: test/e2e/apimachinery/garbage_collector.go (hunk 4 above)
Comment (brahmaroutu) [reply test/e2e/apimachinery/garbage_collector.go]: Fixed.

**Comment by brahmaroutu** [comment]
@sttts Please review. This is documentation only changes and would like to push it through for 1.12 so that we can generate Conformance Document for the releases.

**Comment by spiffxp** [comment]
/lgtm
/approve
I have many style nits around resource naming and style conventions but would prefer to merge and see what the overall doc looks like before focusing on consistency across all test descriptions

**Pull request merged**